### ISSUE TITLE
"c" format should be considered invariant for TimeSpan

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#495]
The "c" format string for **TimeSpan** is [documented](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-timespan-format-strings) as invariant (not culture-sensitive).

I propose to no longer report **MA0011** / **MA0076** when the type is **TimeSpan** and the format string is "c".

**MA0076** is currently not reported for format "c", but this appears to be caused by a small bug as it's also not reported for format "g" (which is not invariant).

```c#
using System;

var ts = TimeSpan.FromHours(2);
Console.WriteLine("Waiting for " + ts.ToString("c") + "..."); // MA0011 should not be reported
Console.WriteLine($"Waiting for {ts:c}..."); // MA0076 should not be reported

Console.WriteLine("Waiting for " + ts.ToString("g") + "..."); // MA0011 should be reported
Console.WriteLine($"Waiting for {ts:g}..."); // MA0076 should be reported
```

**Comment by drieseng** [comment]
Let me know if you want me to create a separate issue for MA0076 issue.
